rfd/XXXX-upgrade-system.md

```diff
@@ -0,0 +1,1222 @@
+---
+authors: Forrest Marshall (forrest@goteleport.com)
+state: draft
+---
+
+# RFD XXXX - Upgrade System
+
+## What
+
+System for automatic upgrades of teleport installations.
+
+
+## Why
```

Comment:
Nit: I would mention Teleport Cloud in the why section, as it's a major driver for this feature (we can't upgrade proxy/auth on tenants where the nodes are out of date).